### PR TITLE
Add --ignore-not-found flag logic to zosfilesBase.handler and update command definitions and en.ts for delete operations

### PR DESCRIPTION
**Pull request by ATorrise**
**What it Does**
This PR implements the `--ignore-not-found` flag for delete operations across DS, USS, zFS, and VSAM by:
- Adding logic in the `ZosFilesBaseHandler` to handle `--ignore-not-found` and suppress errors when a file or data set does not exist (HTTP 404 or IDC3012I).
  - The IDC3012I code is an IBM z/OS error message that indicates a VSAM entry does not exist in the catalog
- Updating the command definitions to include the `--ignore-not-found` flag.
- Modifying `en.ts` to reflect the new flag in descriptions and examples.
 
image

**How to Test**
Verify that the `--ignore-not-found` flag works for all delete operations (DS, USS, zFS, VSAM) by attempting to delete things that don't exist. Delete non-existent files/data sets with and without `--ignore-not-found`.

1. Test commands with --for-sure and --ignore-not-found flag (-fi)

| Command                                                | Type  |
|--------------------------------------------------------|-------|
| `zowe zos-files delete data-set "ibmuser.cntl" -fi`     | DS    |
| `zowe zos-files delete uss-file "/a/ibmuser/testcases" -fi` | USS   |
| `zowe zos-files delete zos-file-system "HLQ.MYNEW.ZFS" -fi` | zFS   |
| `zowe zos-files delete data-set-vsam "ibmuser.AAA.**.FFF" -fi` | VSAM  |

2. Test commands with just the --for-sure flag (-f)

| Command                                                | Type  | Expected Error Message                                          |
|--------------------------------------------------------|-------|-----------------------------------------------------------------|
| `zowe zos-files delete data-set "ibmuser.cntl" -f`      | DS    | "Data set not cataloged"                 |
| `zowe zos-files delete uss-file "/a/ibmuser/testcases" -f` | USS   | "No such file or directory"             |
| `zowe zos-files delete zos-file-system "HLQ.MYNEW.ZFS" -f` | zFS   | "NOT FOUND"         |
| `zowe zos-files delete data-set-vsam "ibmuser.AAA.**.FFF" -f` | VSAM  | "NOT DELETED"       |

**Review Checklist**
I certify that I have:
- [x] I tested my changes.
- [x] I added/updated automated tests.
- [x] I updated the changelog.
- [x] I followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md).

**Additional Comments**
This update is for all delete operations except MDS (which does not currently support this flag)

### CODE REVIEW

**Comment by dkelosky** [comment]
Perhaps it's too late to adjust this, but I wanted to make an observation... `--quite` seems like a pretty generic flag that might have applicability to other command groups besides `zowe files delete` operations.  Should it be reserved for a broader purpose?  

I also came across [this suggested "standards" reference](https://www.gnu.org/prep/standards/html_node/Command_002dLine-Interfaces.html) which links to description:

> ‘quiet’
Used in many programs to inhibit the usual output. Every program accepting ‘--quiet’ should accept ‘--silent’ as a synonym.

Alternatively, a more descriptive command option for this command group could be used `--ignore-not-found`?

File: packages/cli/__tests__/zosfiles/__unit__/delete/zfs/zfs.handler.unit.test.ts
Comment (zFernand0) [inline packages/cli/__tests__/zosfiles/__unit__/delete/zfs/zfs.handler.unit.test.ts:90]: Hey @ATorrise,
I forgot to add the `mockImplementationOnce` back on both tests.
It might be the cause of other tests failing. Sorry about that 😢

**Comment by t1m0thyj** [comment]
> Perhaps it's too late to adjust this, but I wanted to make an observation... `--quiet` seems like a pretty generic flag that might have applicability to other command groups besides `zowe files delete` operations. Should it be reserved for a broader purpose?
> 
> Alternatively, a more descriptive command option for this command group could be used ~`--ignore-not-found`?

I think this is a good point that `--quiet` or `--silent` are pretty generic and could be used for other purposes in the future such as controlling log level.

If it's not too late to change, I support the idea of `--ignore-not-found` which is more explicit and used by some other CLIs like [kube-cli](https://github.com/kubernetes/kubernetes/issues/10828) on their delete commands 😋

File: packages/cli/__tests__/zosfiles/__system__/delete/zfs/__snapshots__/cli.files.delete.zfs.system.test.ts.snap
Comment (zFernand0) [inline packages/cli/__tests__/zosfiles/__system__/delete/zfs/__snapshots__/cli.files.delete.zfs.system.test.ts.snap:1]: Ideally we would not have so many snapshots 😋 
Not asking to change anything at this point since many other tests use them for these success messages 😓

File: packages/cli/__tests__/zosfiles/__system__/delete/ds/__snapshots__/cli.files.delete.ds.system.test.ts.snap
Comment (zFernand0) [inline packages/cli/__tests__/zosfiles/__system__/delete/ds/__snapshots__/cli.files.delete.ds.system.test.ts.snap:40]: is this possibly reflecting a breaking change? 

If someone deletes a dataset with `--rfj`, should they still get the `data.commandResponse` and `data.success` values back?

File: packages/cli/src/zosfiles/delete/zfs/zfs.definition.ts
Comment (zFernand0) [inline packages/cli/src/zosfiles/delete/zfs/zfs.definition.ts:53]: This comment applies to other option definitions.

To be consistent with our [rules for naming commands](https://github.com/zowe/zowe-cli/blob/master/docs/CommandFormatStandards.md#naming-commands) (and options), I think that instead of `ignore-nf`, it should be `inf`
We can keep the `i` if we want people to use `-if` when deleting datasets and files 😋 

```suggestion
            aliases: ["i", "inf"],
```

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (zFernand0) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:56]: To avoid the breaking change with `--rfj`, I think we should do the following:
```suggestion
            commandParameters.response.data.setObj(response);
        } catch (error) {
```

**Review by zFernand0 — CHANGES_REQUESTED** [review]
LGTM! 😋 
left a small change-request to avoid a small breaking change on people processing out `--rfj`

File: packages/cli/__tests__/zosfiles/__system__/delete/ds/__snapshots__/cli.files.delete.ds.system.test.ts.snap
Comment (ATorrise) [reply packages/cli/__tests__/zosfiles/__system__/delete/ds/__snapshots__/cli.files.delete.ds.system.test.ts.snap:40]: is it necessary to keep this in the test? I just removed  it because it didn't seem necessary since im also checking for an error code @zFernand0

Comment (zFernand0) [reply packages/cli/__tests__/zosfiles/__system__/delete/ds/__snapshots__/cli.files.delete.ds.system.test.ts.snap:40]: After adding `commandParameters.response.data.setObj(response);` back, you may need to run system tests again.
And this `data: { }` object should be filled again 😋

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (zFernand0) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:65]: This is fine! 🔥 🐶 🔥 

What about the processWithSession implementations that do not throw an error, but instead return "gracefully" with `success: false` and a proper response on the commandResponse object?

Even though the ZosFilesBaseHandler is [not exported to end users/developers](https://github.com/zowe/zowe-cli/blob/next/packages/cli/src/index.ts), we still need to maintain the behavior for our own APIs.

Here are the ones that I could find quickly:
- List.dataSetsMatchingPattern
- Download.allMembers
- Upload.pathToDataSet

I believe we still need to check for the response.success value and throw and error (in the `try` block)

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Thanks for addressing a few comments so quickly 😋 
I believe that running the whole suite of system tests might be beneficial as there is a slight change that should be caught by other seemingly unrelated system tests. 😓

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (t1m0thyj) [reply packages/cli/src/zosfiles/ZosFilesBase.handler.ts:65]: Changes to this file will affect every command in the `zowe zos-files` group. If the `IDC3012I` error is specific to VSAMs, would it make sense to handle it in `delete/vsam/Vsam.handler.ts` instead?

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (zFernand0) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:52]: I believe a few unit tests might be failing because of the `response.success === false` possibility.

Comment (ATorrise) [reply packages/cli/src/zosfiles/ZosFilesBase.handler.ts:53]: my unit tests arent failing, im confused about what you mean

Comment (t1m0thyj) [reply packages/cli/src/zosfiles/ZosFilesBase.handler.ts:53]: @ATorrise Tests are failing in GH workflows, here's an example run: https://github.com/zowe/zowe-cli/actions/runs/10944131154/job/30385249824

**Review by zFernand0 — COMMENTED** [review]
waiting to review 😋

**Comment by ATorrise** [comment]
zos files system tests are passing

File: packages/cli/CHANGELOG.md
Comment (zFernand0) [inline packages/cli/CHANGELOG.md:6]:
```suggestion
- Enhancement: Add `--ignore-not-found` flag logic to zosfilesBase.handler and update command definitions and en.ts for delete operations. [#2254](https://github.com/zowe/zowe-cli/pull/2254)
```

File: packages/cli/__tests__/zosfiles/__unit__/upload/ftds/__snapshots__/FileToDataSet.handler.unit.test.ts.snap
Comment (zFernand0) [inline packages/cli/__tests__/zosfiles/__unit__/upload/ftds/__snapshots__/FileToDataSet.handler.unit.test.ts.snap:5]: not something to resolve on this PR, but I'm curious if we can avoid the use of snapshots for empty strings or undefined/null values 😋

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (zFernand0) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:50]: should this be moved below the call to `processWithSession`? 
if we don't, some progress bars may terminate before they are supposed to 😅

**Review by zFernand0 — DISMISSED** [review]
LGTM! 😋 

left a few small comments/questions, but nothing that should stop the PR from being merged if it gets enough approvals 🥳

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Looks pretty good, thanks @ATorrise!

Will approve once this comment is addressed: https://github.com/zowe/zowe-cli/pull/2254#discussion_r1803211668

File: packages/cli/__tests__/zosfiles/__system__/delete/vsam/cli.files.delete.vsam.system.test.ts
Comment (awharn) [inline packages/cli/__tests__/zosfiles/__system__/delete/vsam/cli.files.delete.vsam.system.test.ts:123]: The portion of this test that creates the VSAM to delete was removed, and this test does not pass.

**Review by awharn — DISMISSED** [review]
Found one issue that should be corrected before merge.

File: packages/cli/src/zosfiles/delete/vsam/Vsam.handler.ts
Comment (awharn) [inline packages/cli/src/zosfiles/delete/vsam/Vsam.handler.ts:35]: Please add the original error to causeErrors.

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (awharn) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:66]: Please add the original error to causeErrors.

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (awharn) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:64]: We should check if an ImperativeError is thrown and just rethrow it. Otherwise we might have an ImperativeError in an ImperativeError.

File: packages/cli/src/zosfiles/delete/vsam/Vsam.handler.ts
Comment (awharn) [reply packages/cli/src/zosfiles/delete/vsam/Vsam.handler.ts:35]: Actually, we should probably check if it is an Imperative Error and if so, rethrow it.

File: packages/cli/CHANGELOG.md
Comment (anaxceron) [inline packages/cli/CHANGELOG.md:5]: Suggesting to change this to something like: "Add the `--ignore-not-found` flag to avoid error messages potentially interrupting a script for automated batch processes. The flag bypasses warning prompts to confirm delete actions."

**Review by anaxceron — COMMENTED** [review]
Suggesting some edits to the changelog.

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (awharn) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:70]: I believe all of this code should be how it was prior to the changes, with a try/catch added around it and the special handling for a 404 with ignoreNotFound. Because of the order we are running this now, the daemon is never getting data back because we are not setting data, and the daemon is waiting for data that never comes. This causes the daemon to loop infinitely and freeze, which causes the tests to fail due to timing out.

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (ATorrise) [reply packages/cli/src/zosfiles/ZosFilesBase.handler.ts:66]: i guess i didnt really know what to do here because response isnt accessible here because of the catch block so i cant really just like revert back @awharn

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (t1m0thyj) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:62]: The integration tests are still failing because the order of these statements has been switched around. Restoring the original order should fix them 😋 
1. `commandParameters.response.progress.endBar();`
2. `if (response.commandResponse) ...`
3. `commandParameters.response.data.setObj(response);`
4. `if (!response.success && response.commandResponse) ...`

**Review by anaxceron — APPROVED** [review]
Good changelog, thank you @ATorrise!

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

File: packages/cli/__tests__/zosfiles/__system__/delete/uss/cli.files.delete.file.system.test.ts
Comment (t1m0thyj) [inline packages/cli/__tests__/zosfiles/__system__/delete/uss/cli.files.delete.file.system.test.ts:135]: Should this test have a second step to match the related test for data sets?
`// Now delete the USS file, verify no output in ignore-not-found mode`

File: packages/cli/__tests__/zosfiles/__system__/delete/vsam/cli.files.delete.vsam.system.test.ts
Comment (t1m0thyj) [inline packages/cli/__tests__/zosfiles/__system__/delete/vsam/cli.files.delete.vsam.system.test.ts:140]: Same question as for USS above - should we attempt the deletion a 2nd time and make sure it still works with `--inf` flag?

File: packages/cli/__tests__/zosfiles/__system__/delete/zfs/cli.files.delete.zfs.system.test.ts
Comment (t1m0thyj) [inline packages/cli/__tests__/zosfiles/__system__/delete/zfs/cli.files.delete.zfs.system.test.ts:126]: Same question as for USS above - should we attempt the deletion a 2nd time and make sure it still works with `--inf` flag?

File: packages/cli/__tests__/zosfiles/__unit__/create/zfs/zfs.handler.unit.test.ts
Comment (t1m0thyj) [inline packages/cli/__tests__/zosfiles/__unit__/create/zfs/zfs.handler.unit.test.ts:135]: Not a request for changes, just a note that in the future it may be helpful to use `mockHandlerParameters` method from the cli-test-utils package to reduce the amount of code needed to mock the handler parameters object 😋

File: packages/cli/__tests__/zosfiles/__unit__/mount/fs/fs.handler.unit.test.ts
Comment (t1m0thyj) [inline packages/cli/__tests__/zosfiles/__unit__/mount/fs/fs.handler.unit.test.ts:1]: Curious why changes were needed in this file as well as the one above (`DataSetMatching.handler.unit.test.ts`), since it doesn't seem related to the rest of the PR. It's probably ok since these changes are to tests.

File: packages/cli/src/zosfiles/ZosFilesBase.handler.ts
Comment (t1m0thyj) [inline packages/cli/src/zosfiles/ZosFilesBase.handler.ts:64]: Not a request for changes but please be careful about removing comments - some other ones in this file probably aren't very meaningful like "print out the response" so I understand removing them, but more verbose comments like this one may contain valuable information.

File: packages/cli/src/zosfiles/delete/vsam/Vsam.handler.ts
Comment (t1m0thyj) [inline packages/cli/src/zosfiles/delete/vsam/Vsam.handler.ts:38]: The `mMessage` property is mostly likely undefined if the error is not an instance of `ImperativeError`. Should we do the following to match the base handler?
```suggestion
                msg: error.message,
```

File: packages/zosfiles/CHANGELOG.md
Comment (t1m0thyj) [inline packages/zosfiles/CHANGELOG.md:1]: Please remove changelog update from this file since no user-facing changes were made to the zos-files SDK, only a formatting change

**Review by t1m0thyj — COMMENTED** [review]
Looks pretty good, thanks @ATorrise! Left a few comments

**Review by anaxceron — APPROVED** [review]

**Comment by ATorrise** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=zowe_zowe-cli&pullRequest=2254) **Quality Gate failed**
> Failed conditions  [4.2% Duplication on New Code](https://sonarcloud.io/component_measures?id=zowe_zowe-cli&pullRequest=2254&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=zowe_zowe-cli&pullRequest=2254)

@zFernand0 @t1m0thyj not sure what to do about this?

**Comment by zFernand0** [comment]
> > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=zowe_zowe-cli&pullRequest=2254) **Quality Gate failed**
> > Failed conditions  [4.2% Duplication on New Code](https://sonarcloud.io/component_measures?id=zowe_zowe-cli&pullRequest=2254&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
> > [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=zowe_zowe-cli&pullRequest=2254)
> 
> `@zFernand0` @t1m0thyj not sure what to do about this?

I wouldn't worry too much about the duplication.
of course there could be some of it removed, but it might make the code a bit harder to follow.

If we do plan to reduce duplication, I guess I'd prefer to do it on a major release boundary, as part of our cleanup efforts 😋

File: packages/cli/CHANGELOG.md
Comment (t1m0thyj) [inline packages/cli/CHANGELOG.md:5]: I'm not sure that FNF is a commonly known acronym, should we spell it out?

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @ATorrise for the new feature and thorough system tests!

**Pull request merged**
